src/snapshot_tests/scene_test.rs

```diff
@@ -0,0 +1,192 @@
+
+        self.check_for_unused_snapshots(&snapshots)?;
+        Ok(())
+    }
+
+    pub fn generate_snapshots(&self) -> Result<Vec<Snapshot>> {
+        let snapshots = self
+            .timestamps
+            .iter()
+            .cloned()
+            .map(|pts| self.snapshots_for_pts(pts))
+            .collect::<Result<Vec<_>>>()?;
+
+        Ok(snapshots.into_iter().flatten().collect())
+    }
+
+    fn snapshots_for_pts(&self, pts: Duration) -> Result<Vec<Snapshot>> {
+        let mut frame_set = FrameSet::new(pts);
+        for input in self.inputs.iter() {
+            let input_id = InputId(NodeId(input.name.into()));
+            let frame = Frame {
+                data: input.data.clone(),
+                resolution: input.resolution,
+                pts,
+            };
+            frame_set.frames.insert(input_id, frame);
+        }
+
+        let outputs = self.renderer.render(frame_set)?;
+        let output_specs = &self.scene.outputs;
+        let mut snapshots = Vec::new();
+
+        for spec in output_specs {
+            let output_frame = outputs.frames.get(&spec.output_id).unwrap();
+            let new_snapshot = frame_to_rgba(output_frame);
+            snapshots.push(Snapshot {
+                test_name: self.test_name.to_owned(),
+                output_id: spec.output_id.clone(),
+                pts,
+                resolution: output_frame.resolution,
+                data: new_snapshot,
+            });
+        }
+
+        Ok(snapshots)
+    }
+
+    fn check_for_unused_snapshots(&self, snapshots: &[Snapshot]) -> Result<()> {
+        let snapshot_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .join(SNAPSHOTS_DIR_NAME)
+            .join(self.test_name);
+        let parent_path = snapshot_path.parent().unwrap();
+
+        for entry in fs::read_dir(parent_path).unwrap() {
+            let entry = entry.unwrap();
+            if !entry.file_name().to_string_lossy().ends_with(".bmp") {
+                continue;
+            }
+            if !snapshots.iter().any(|s| s.save_path() == entry.path()) {
+                return Err(SceneTestError::UnusedSnapshot(entry.path()).into());
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Snapshot {
+    pub test_name: String,
+    pub output_id: OutputId,
+    pub pts: Duration,
+    pub resolution: Resolution,
+    pub data: Vec<u8>,
+}
+
+impl Snapshot {
+    pub fn save_path(&self) -> PathBuf {
+        let snapshots_path = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join(SNAPSHOTS_DIR_NAME);
+
+        let out_file_name = format!(
+            "{}_{}_{}.bmp",
+            self.test_name,
+            self.pts.as_millis(),
+            self.output_id
+        );
+        snapshots_path.join(out_file_name)
+    }
+}
+
+#[derive(Debug)]
+pub enum SceneTestError {
+    SnapshotNotFound(Snapshot),
+    Mismatch(Snapshot),
+    UnusedSnapshot(PathBuf),
+    OutputNotFound(&'static str),
+    TooManyOutputs {
```

Comment:
UnknownOutput
That issue means that you received some input you shouldn't not that there is to many outputs